### ISSUE TITLE
Synk code scan warning on OidcConfiguration

### ISSUE THREAD

**Issue by tadayosi** [hawtio/hawtio#3318]
`snyk code test` shows this warning:
```
 ✗ [Medium] Server-Side Request Forgery (SSRF) 
   Path: hawtio-system/src/main/java/io/hawt/web/auth/oidc/OidcConfiguration.java, line 573 
   Info: Unsanitized input from data from a remote resource flows into org.apache.http.client.methods.HttpGet, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.
```
```
#  571|       private JSONObject fetchJSON(URL url) {
#  572|           try {
#  573|->             HttpGet get = new HttpGet(url.toURI());
#  574|               LOG.info("Fetching data: {}", get.getRequestLine());
#  575|               try (CloseableHttpResponse res = httpClient.execute(get)) {
```
It's not as easy to fix as it first appears. Does anyone know how to remove the warning?  @mmelko @grgrzybek

**Comment by grgrzybek** [comment]
Nice... So according to SNYK, we can only use hardcoded URLs?

**Comment by grgrzybek** [comment]
We can replace `new HttpGet(url.toURI())` with `new HttpGet(validate(url.toURI()))`, where `validate()` will just return the same URL - I wonder if we can trick it ;)

But seriously - how about https://github.com/hawtio/hawtio/blob/hawtio-4.0.0-RC1/hawtio-system/src/main/java/io/hawt/web/proxy/ProxyServlet.java#L267 ?

**Comment by tadayosi** [comment]
> Nice... So according to SNYK, we can only use hardcoded URLs?

It's not that simple. We instead just want to satisfy `snyk code test`, and hardcoding URLs might not be the only solution.

Right now, the result shows:
```
$ snyk code test

Testing /home/tasato/projects/hawtio/hawtio ...

 ✗ [Low] Use of Hardcoded Credentials 
   Path: hawtio-system/src/main/java/io/hawt/system/Authenticator.java, line 126 
   Info: Do not hardcode credentials in code.

 ✗ [Low] Use of Hardcoded Credentials 
   Path: hawtio-system/src/main/java/io/hawt/web/auth/AuthenticationThrottler.java, line 48 
   Info: Do not hardcode credentials in code.

 ✗ [Low] Use of Password Hash With Insufficient Computational Effort 
   Path: hawtio-system/src/main/java/io/hawt/web/tomcat/TomcatUserDatabaseLoginContext.java, line 56 
   Info: The MD5 hash (used in org.apache.commons.codec.digest.DigestUtils.md5Hex) is insecure. Consider changing it to a secure hash algorithm

 ✗ [Low] Use of Password Hash With Insufficient Computational Effort 
   Path: hawtio-system/src/main/java/io/hawt/web/tomcat/TomcatUserDatabaseLoginContext.java, line 64 
   Info: The SHA hash (used in org.apache.commons.codec.digest.DigestUtils.shaHex) is insecure. Consider changing it to a secure hash algorithm

 ✗ [Medium] Origin Validation Error 
   Path: hawtio-system/src/main/java/io/hawt/web/filters/CORSFilter.java, line 96 
   Info: CORS policy "*" might be too permissive. This allows malicious code on other domains to communicate with the application, which is a security risk

 ✗ [Medium] Server-Side Request Forgery (SSRF) 
   Path: hawtio-system/src/main/java/io/hawt/web/auth/oidc/OidcConfiguration.java, line 573 
   Info: Unsanitized input from data from a remote resource flows into org.apache.http.client.methods.HttpGet, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.


✔ Test completed

Organization:      application-services-red-hat-camel-hawtio
Test type:         Static code analysis
Project path:      /home/tasato/projects/hawtio/hawtio

Summary:

  6 Code issues found
  2 [Medium]   4 [Low] 
```
so `ProxyServlet` is forgiven and the other warnings are mostly known as false positives. We only need to fix `OidcConfiguration` right now.

**Comment by tadayosi** [comment]
Fixed with #3319
